.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+name: Changesets
+on:
+  workflow_dispatch:
+    inputs:
+      level:
+        description: 'Release type'
+        required: true
+        default: 'preview'
+        type: choice
+        options:
+          - preview
+          - general availability
+env:
+  CI: true
+  PNPM_CACHE_FOLDER: .pnpm-store
+jobs:
+  version:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: setup node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
+      - name: install pnpm
+        run: npm i pnpm@latest -g
+      - name: Setup npmrc
+        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > .npmrc
```

Comment:
How/where is this secret defined please?

Reply:
This is a secret defined for the action, that can be at repository or organization level. https://docs.github.com/en/actions/security-guides/encrypted-secrets